datasets/datasets.py

```diff
@@ -70,36 +71,38 @@ def create_dataset(file_object):
         file = file_object.file
         filename = file_object.filename
 
-    # if user does not select file, the browser also
-    # submits an empty part without filename
-    if filename == "":
-        raise BadRequest("No selected file.")
-
-    # generate a dataset name from filename
-    name = generate_name(filename)
-
-    try:
-        # reads file into a DataFrame
-        df = read_into_dataframe(file, filename)
-    except Exception:
-        # if read fails, then uploads raw file
-        file.seek(0, SEEK_SET)
-        save_dataset(name, file, metadata={"original-filename": filename})
-        return {"name": name, "filename": filename}
-
-    columns = df.columns.values.tolist()
-    featuretypes = infer_featuretypes(df)
-
-    metadata = {
-        "columns": columns,
-        "featuretypes": featuretypes,
-        "original-filename": filename,
-        "total": len(df.index),
-    }
-
     file.seek(0, SEEK_SET)
-    # uses PlatIAgro SDK to save the dataset
-    save_dataset(name, file, metadata=metadata)
+    with NamedTemporaryFile() as buffer:
+        shutil.copyfileobj(file, buffer)
+
+        # if user does not select file, the browser also
+        # submits an empty part without filename
+        if filename == "":
+            raise BadRequest("No selected file.")
+
+        # generate a dataset name from filename
+        name = generate_name(filename)
+
+        try:
+            # reads file into a DataFrame
+            df = read_into_dataframe(buffer, filename)
+        except UnicodeDecodeError:
```

Comment:
Caso o arquivo não for do tipo que possa ser lido como um dataframe então ele é colocado em sua forma 'crua' no storage, essa parte é útil quando se deseja 'upar' arquivos do tipo imagem por exemplo